### ISSUE TITLE
Primary key issues with really long table names

### ISSUE THREAD

**Issue by jarednorman** [rails/rails#31625]
### Steps/Actual Behaviour

Example app demonstrating the issue [here](https://github.com/jarednorman/rails_bug_bad_id) with instructions in the README. Here's the full steps:

- Create a new Rails 5.1 app (I used Rails 5.1.4, Ruby 2.4.1) and postgres as the db (I'm running 9.6).
- Create a table with a really long name (long enough that it will get truncated when generating the name for the seq for the primary key)
- Examine the schema.rb, you'll see that rather than `id: :serial`, you'll have something like that looks like:

  ```ruby
  create_table "sorry_i_didnt_name_the_table_that_caused_this_problem_for_me", id: :bigint, default: -> { "nextval('sorry_i_didnt_name_the_table_that_caused_this_problem_fo_id_seq'::regclass)" }, force: :cascade do |t|
    # ...
  end
  ```

  That's technically the correct default (that's the correct seq) but it seems ActiveRecord has forgotten that fact, and is treating like it's a custom default.
- Drop the database and run `bundle exec rake db:setup`. It will error out with this message (full trace is [here](https://github.com/jarednorman/rails_bug_bad_id#how-to-reproduce)):

  ```
  ActiveRecord::StatementInvalid: PG::UndefinedTable: ERROR:  relation "sorry_i_didnt_name_the_table_that_caused_this_problem_fo_id_seq" does not exist
  ```

### Expected behavior

The schema.rb should reflect that the we're using normal serial ID's here:

```ruby
  create_table "sorry_i_didnt_name_the_table_that_caused_this_problem_for_me", id: :serial, force: :cascade do |t|
    # ...
  end
```

This would fix the issue that the schema produced cannot be loaded. I consider the `UndefinedTable` error to be expected given what was in the schema.

### System configuration
**Rails version**: `5.1.4`
**Ruby version**: `2.4.1`
**Postgresql version**: `9.6.6`

**Comment by kamipo** [comment]
This has been fixed in #28339 and backported in af9c1707ad49eb421cf76c1367da3efbf3c39aea.

**Comment by jarednorman** [comment]
Thanks! Sorry I failed to find the original bug.
